rules/provider.go

```diff
@@ -43,11 +44,35 @@ var manualDeepCheckRules = []Rule{
 	awsrules.NewAwsLaunchConfigurationInvalidImageIDRule(),
 }
 
+// AllRulesMap returns map of rules indexed by name
+func AllRulesMap() map[string]Rule {
+	ret := map[string]Rule{}
+	for _, rule := range append(DefaultRules, deepCheckRules...) {
+		ret[rule.Name()] = rule
+	}
+	return ret
+}
+
 // NewRules returns rules according to configuration
-func NewRules(c *tflint.Config) []Rule {
+func NewRules(c *tflint.Config) ([]Rule, error) {
```

Comment:
Sorry, I'm planning to make a major change to the implementation of this provider, so I don't encourage you to change `NewRules`. Perhaps we should implement a method that simply returns a list of rule names.

This allows for the possibility that the Provider will be provided by the user as a plug-in in the future. Since `NewRules` does not return all rules, we cannot check whether the passed rule name exists.

Reply:
See #491

Reply:
I almost finished [error processing](https://github.com/golang/go/wiki/ErrorValueFAQ) code. :D

Makes total sense to check options separately before `NewRules` is called. Now that I know the codebase, I can try to do this.